### PR TITLE
Make click events on status blocks work if 'workspace_buttons no' is set...

### LINKED ISSUE #1430: i3bar does not send click events when workspace buttons are hidden
[Originally reported by i3-new-ticket]
(When workspace buttons are hidden 

```
bar {
	...
	workspace_buttons no
}
```

i3bar does not send click events even if the appropriate header block is sent

```
{"version": 1, "click_events": true}
```

click events can be received if `workspace_buttons yes` is set in `bar` block.

### PR DESCRIPTION
**Pull request by Airblader**
....

1. Always subscribe to click events for i3bar.
2. Exit the click event handler if no current workspace was found only after clicks on status blocks have been handled.

fixes #1430

### CODE REVIEW

File: i3bar/src/child.c [hunk 1]
```diff
@@ -520,33 +520,35 @@ void child_click_events_key(const char *key) {
  *
  */
 void send_block_clicked(int button, const char *name, const char *instance, int x, int y) {
-    if (child.click_events) {
```
Comment (Airblader) [inline i3bar/src/child.c]: `git diff` is bugging out here. This is just an early return to get rid of one level of intendation.

File: i3bar/src/xcb.c [hunk 2]
```diff
@@ -364,6 +352,18 @@ void handle_button(xcb_button_press_event_t *event) {
         x = original_x;
     }
 
+    /* TODO: Move this to extern get_ws_for_output() */
```
Comment (Airblader) [inline i3bar/src/xcb.c]: The actual fix. Since i3bar doesn't subscribe to the workspaces event if no workspace are to be drawn, it will not find a `cur_ws` and return. I figured moving this down to after the check for status block clicks is more reasonable than subscribing to the event.

File: i3bar/src/xcb.c [hunk 3]
```diff
@@ -1460,10 +1460,8 @@ void reconfig_windows(bool redraw_bars) {
              * BUTTON_PRESS, to handle clicks on the workspace buttons
              * */
             values[2] = XCB_EVENT_MASK_EXPOSURE |
-                        XCB_EVENT_MASK_SUBSTRUCTURE_REDIRECT;
-            if (!config.disable_ws) {
-                values[2] |= XCB_EVENT_MASK_BUTTON_PRESS;
-            }
+                        XCB_EVENT_MASK_SUBSTRUCTURE_REDIRECT |
```
Comment (Airblader) [inline i3bar/src/xcb.c]: Also part of the fix; at this point we don't know yet whether the child will request click events, so we need to subscribe to them in any case. This will cause a merge conflict to `i3/next`, but a trivial one.

File: i3bar/src/child.c (hunk 1 above)
Comment (stapelberg) [inline i3bar/src/child.c]: Can you move this to a separate commit, or leave it out of this pull request entirely? That’d make it easier to review the pull request itself.

**Comment by stapelberg** [comment]
Looks good, aside from the one comment about making the commit diff easier to understand.

**Comment by Airblader** [comment]
@stapelberg Sure. Since I can't change commits postum, you'll have to reject this PR and I'll submit a new one.

**Comment by stapelberg** [comment]
@Airblader You can git push --force, can’t you?

**Comment by Airblader** [comment]
@stapelberg I can, but I'm not sure if this will affect the PR. Let's find out!

**Comment by Airblader** [comment]
Cool, it worked. :)

**Comment by Airblader** [comment]
@stapelberg Don't forget to merge this up to `i3/next` :)

**Comment by stapelberg** [comment]
Already did. Please send pull requests for next directly in the future.

**Comment by Airblader** [comment]
@stapelberg This was a bug fix, which according to the guide should go to `i3/master`, no?

**Comment by stapelberg** [comment]
Theoretically yes, but I’d prefer it if I did the backports myself when merging. From a developer point of view, you should only ever have to care about `next`.

**Comment by Airblader** [comment]
Ah, okay. Willdo :)
